packages/agreement-process/src/services/agreementActivationProcessor.ts

```diff
@@ -123,32 +130,67 @@ export async function createActivationEvent(
       updatedAgreementSeed
     );
 
-    return toCreateEventAgreementActivated(
-      {
-        ...updatedAgreement,
-        contract: apiAgreementDocumentToAgreementDocument(agreementContract),
-      },
-      agreement.metadata.version,
-      correlationId
-    );
-  } else {
-    if (authData.organizationId === agreement.data.producerId) {
-      return toCreateEventAgreementUnsuspendedByProducer(
-        updatedAgreement,
-        agreement.metadata.version,
-        correlationId
-      );
-    } else if (authData.organizationId === agreement.data.consumerId) {
-      return toCreateEventAgreementUnsuspendedByConsumer(
-        updatedAgreement,
+    return [
+      toCreateEventAgreementActivated(
+        {
+          ...updatedAgreement,
+          contract: apiAgreementDocumentToAgreementDocument(agreementContract),
+        },
         agreement.metadata.version,
         correlationId
-      );
-    } else {
-      throw genericError(
-        `Unexpected organizationId ${authData.organizationId} in activateAgreement`
-      );
-    }
+      ),
+    ];
+  } else {
+    return match([authData.organizationId, updatedAgreement.state])
+      .with([agreement.data.producerId, agreementState.active], () => [
+        toCreateEventAgreementUnsuspendedByProducer(
+          updatedAgreement,
+          agreement.metadata.version,
```

Comment:
If I see properly `agreement` parameter it's used only two reason retrieve `version`and `suspendedByPlatform` flag; the other usages like reading `agreement.data.producerId` could be executed using `updatedAgreement` because contains the same data of `agreement`, in fact it merged in the line https://github.com/pagopa/interop-be-monorepo/pull/578/files#diff-71db2d22095ae81787fc9ee7ffb2606a4a832b763fb13e3c5d19646434e3d951R1003 just before invoking this method.
I suggest to using only parameter necessary to check differences between orginal version of agreement and submitted, in this way we avoid ambiguity between  agreement submitted and not in this function execution.